src/float/acos.rs

```diff
@@ -4,24 +4,65 @@
 //! <https://math.stackexchange.com/questions/2908908/express-arccos-in-terms-of-arctan>
 
 use super::F32;
+use core::f32::consts::PI;
 
 impl F32 {
     /// Computes `acos(x)` approximation in radians in the range `[0, pi]`.
     pub(crate) fn acos(self) -> Self {
-        ((Self::ONE - self * self).sqrt() / self).atan()
+        if self > 0.0 {
+            ((Self::ONE - self * self).sqrt() / self).atan()
+        } else if self == 0.0 {
+            F32(PI / 2.0)
```

Comment:
For consistency with the other ops I'd suggest:

```suggestion
            Self(PI / 2.0)
```